lib/ace/ext/papeeria/katex-previewer.coffee

```diff
@@ -20,18 +19,21 @@ define((require, exports, module) ->
             $("body").append(span)
 
             katex = require("ace/ext/katex")
-            katexInitialized = true
             return
 
         editor.commands.addCommand({
             name: "previewLaTeXFormula",
             bindKey: {win: "Alt-p", mac: "Alt-p"},
             exec: (editor) ->
-                if not katexInitialized
+                unless katex?
                     initKaTeX()
                 selectedText = editor.getSelectedText()
-                katex.render(selectedText, $("#formula")[0])
-                return
+                try
+                    katex.render(selectedText, $("#formula")[0])
+                catch e
+                    $("#formula").text("Can't parse selected text!")
+                finally
```

Comment:
No `return` could be harmful, as CoffeeScript automatically returns the last statement in the function. Potentially that value can be used inside ACE, and I don't think we want that.